### DISCUSSION TITLE
How to setup Dynamic Hosts and Allowed Redirect URLs

### DISCUSSION THREAD

**Discussion by ghost** [codatio/codat-docs#659] [Q&A] [answered]
Hi, I'm trying to setup a dynamic redirect so my users can be directed a single endpoint depending on which environment they are using. I have tried following the docs https://docs.codat.io/auth-flow/customize/set-up-redirects#dynamic-hosts 

- Set Authorization Complete Redirection Url to `https://www.{domain}/success?statusCode={statusCode}&companyId={companyId}&errorMessage={errorMessage}`
- Set Allowed redirect URLs to `http://localhost:5001`
- Set Allowed redirect URLs to `https://test.app.com`

Then when navigating to `https://link.codat.io/client/X?user=X&domain=http://localhost:5001` I am met with the `400 Invalid redirect URL` error.

Is there a specific format these Allowed Redirect URLs are supposed to be in? It won't let me save just the domain.

Also it's quite confusing as in the settings panel the _Authorization Complete Redirection Url_ is just labelled as _Redirect URL -> URL_.

**Comment by mcclowes** [comment]
If those are your settings, then `https://link.codat.io/client/X?user=X&domain=http://localhost:5001` would redirect you to:

```https://www.http://localhost:5001/success...```

Which isn't a valid URL.

Try:
```https://{domain}/success?statusCode={statusCode}&companyId={companyId}&errorMessage={errorMessage}```

And navigate to:
```https://link.codat.io/client/XYZ?domain=localhost:5001```

Also not 100% sure off the top of my head if the colon in `localhost:5001` would need encoding or anything

**Comment by ghost** [comment, marked as answer]
Just tried adding `localhost` and `localhost:5001` on it's own but was met with this error on the field.

<img width="440" alt="Screenshot 2023-08-10 at 09 40 18" src="https://github.com/codatio/codat-docs/assets/138530684/18d58efe-4955-4973-9e4a-addec8984c68">

Although that has helped me find the solution.

**Authorization Complete Redirection URL**
`{domain}/accountingSuccess?statusCode={statusCode}&companyId={companyId}&errorMessage={errorMessage}`

**Allowed redirect URLs**
http://localhost:5001
https://test.app.com
